### DISCUSSION TITLE
Remember sign in

### DISCUSSION THREAD

**Discussion by janbaykara** [supabase/supabase#3122] [Questions]
Hi all. I've recently set up a supabase-driven project and it's going great!

The one issue I'm having is with persistent user login. Some users are reporting that they often have to log back in.

I've checked the docs and codebase and can't find any examples of 'remember login'-style functionality, and not really sure what that would entail. [Supabase UI includes a checkbox but it's not wired up to anything.](https://github.com/supabase/ui/blob/master/src/components/Auth/Auth.tsx#L297)

I get the sense that users, once logged in, have some kind of session refresh token, but I'm not sure if this is working correctly, needs additional mechanics to make it happen, or something else.

Has anyone managed to make persistent sign-in work? Could you explain how it is meant to work and how Supabase/GoTrue currently does this?

**Comment by dshukertjr** [comment]
Hi @janbaykara!

Supabase sessions should be persisted by default, so this might be a bug. Which language's client are you using and what is the version that you are using?

**Comment by VivekDahiya06** [comment]
Hi @dshukertjr!
I'm using Supabase with MakerKit setup (Next.js + supabase + turborepo) and in that after signin( I use email with password method for signIn ) I check it redirects me to email and when I check my session via supabase.auth.getSession() it returns "null". I don't know why the session remains null even after signIn. Can you give any suggestions ?
